ci/datamgr.py

```diff
@@ -645,38 +617,84 @@ def impala(data_dir):
             port=env.hdfs_port,
             user=env.hdfs_user,
         ),
-        pool_size=CPU_COUNT,
+        pool_size=URLLIB_DEFAULT_POOL_SIZE,
     )
 
-    # validate our environment before performing possibly expensive operations
-    impala_raise_if_cannot_write_to_hdfs(con, env)
+    fs = fsspec.filesystem("file")
+    data_files = fs.find(data_dir)
```

Comment:
Could you filter the data files here so the set comprehension get simpler below?